### DISCUSSION TITLE
issue: Error message isn't updated correctly with yup and array

### DISCUSSION THREAD

**Discussion by val2207** [react-hook-form/react-hook-form#7428] [Q&A]
### Version Number

7.22.5

### Codesandbox/Expo snack

https://codesandbox.io/s/react-hook-form-array-fields-forked-rxhwk?file=/src/index.js

### Steps to reproduce

1: Open command line to see logs
2: Click on "add friends". You will see on console that there is an error message "friends field must have at least 2 items" and isValid at false which is correct.  :)
3: Click on "add friends" one more time and fill all fields. Now isValid is at true but error message is still there "friends field must have at least 2 items"  which is not correct.  :(

Errors seems never updated with yup and an array (maybe with other validation libs too I didn't tested). In my personal code I have in addition with a "min" validation a custom validation everything is working correctly with isValid attribute but again not with Errors object. 

### Expected behaviour

Errors object must be correctly updated with Array and Yup.

### What browsers are you seeing the problem on?

Chrome

### Relevant log output

```shell
isValid: false 
{}
VM40614:85 isValid: false 
{}
VM40614:85 isValid: false 
{friends: {…}}
friends: {message: 'friends field must have at least 2 items', type: 'min', ref: undefined}
[[Prototype]]: Object
VM40614:85 isValid: false 
{friends: {…}}
VM40614:85 isValid: true 
{friends: {…}}
friends: {message: 'friends field must have at least 2 items', type: 'min', ref: undefined}
[[Prototype]]: Object
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment]
Thanks for the detailed issue report. react hook form focus on field level error report (isolate re-render and enhance computation). With field array group error, this is difficult as it results {} vs [] as a result of the error, and each action in hook form (append, prepend) doesn't trigger validation (this is something we can discuss which can be a good feature to add) however, it also comes with a tradeoff, as it would overwrite array index error, again {} vs []. 

## solution 

if you need field array level of validation, you can use `trigger` at `useEffect` level with `fields` as dep.

**Reply by val2207** [reply]
Ok thank you for the clarification. I agree that for fields inside field array we need to use a trigger to update the validation of field array if it's needed. 

I just don't understand when you say each action in hook form doesn't trigger validation. Here in my example if insert or remove is called yup validation is called too and isValid attribute is updated. This mean to me that a validation is trigger ?

**Reply by bluebill1049** [reply]
> Here in my example if insert or remove is called yup validation is called too and isValid attribute is updated. This mean to me that a validation is trigger ?

you are right we do determine the isValid `formState`, however not error state update, because there is no associated field input, that go back to the {} vs [], if we update the errors from [] to {} then it will lost all the field level errors

**Reply by val2207** [reply]
Ok I see and if we do a full validation of every children fields to not lost all field level errors it goes against the idea of isolate re-render. Sad situation... I will keep the ticket as unanswered few days maybe someone will have an idea. For the moment I don't have any.
